packages/postcss-merge-rules/src/index.js

```diff
@@ -166,11 +167,24 @@ function isConflictingProp(propA, propB) {
   if (a.base !== b.base) {
     return false;
   }
+
   // Conflict if rest-count mismatches
   if (a.rest.length !== b.rest.length) {
     return true;
   }
 
+  /* Do not merge conflicting border properties */
+  if (a.base === 'border') {
```

Comment:
I find this function uses some non-obvious terminology. `base` is the first part of the property name excluding the vendor prefix. `rest` all the rest of the name.
I could not find an alternative to using a special case, CSS syntax is just not that regular. I think  in fact `conflict if  count mismatches or if every part is the same` is only true for the `margin- ` and `padding-` properties.